runtime/lib/component/dispatcher.js

```diff
@@ -0,0 +1,103 @@
+var logger = require('logger')('dispatcher')
+
+var config = require('/etc/yoda/component-config.json')
+
+/**
+ * Event/Command Dispatcher.
+ */
+class Dispatcher {
+  constructor (runtime) {
+    this.runtime = runtime
+    this.component = runtime.component
+
+    this.config = this.validateConfig(config)
+  }
+
+  /**
+   *
+   * @private
+   * @param {any} config
+   */
+  validateConfig (config) {
+    if (typeof config !== 'object') {
+      throw new Error('Invalid component-config.json')
+    }
+    var ret = { interception: {} }
+    if (typeof config.interception !== 'object') {
+      throw new Error('config.interception is not an object.')
+    }
+    Object.keys(config.interception).forEach(key => {
+      var match = key.split('.', 2)
+      var componentName = match[0]
+      if (this.runtime.componentLoader.registry[componentName] == null) {
+        throw new Error(`Unknown component(${componentName}) on component-config`)
+      }
+      var targets = config.interception[key]
+      if (!Array.isArray(targets)) {
+        throw new Error(`Unexpected value on component-config.interception.${key}`)
+      }
+      var result = targets.map((it, idx) => {
+        if (typeof it !== 'string') {
+          throw new Error(`Unexpected value on component-config.interception.${key}.${idx}`)
+        }
+        var match = it.split('.', 2)
+        var componentName = match[0]
+        if (this.runtime.componentLoader.registry[componentName] == null) {
+          throw new Error(`Unknown component(${componentName}) on component-config`)
+        }
+        return { component: componentName, method: match[1] }
+      })
+      ret.interception[key] = result
+    })
+
+    return ret
+  }
+
+  /**
+   * Delegates runtime/component event to interceptors.
+   * @param {string} event - event name to be handled
+   * @param {any[]} args - arguments of the event
+   * @returns {false|any} return false if there is no interceptor responds. Anything truthy otherwise.
+   */
+  delegate (event, args) {
```

Comment:
Event source is not checked, in some cases,  component will emit wrong source, ex: `turen` could delegate the event which belongs to `runtime`, is this secure enough?